packages/cardano-services/src/StakePool/DbSyncStakePoolProvider/DbSyncStakePool.ts

```diff
@@ -160,6 +165,9 @@ export class DbSyncStakePoolProvider extends DbSyncProvider implements StakePool
             this.#builder.queryPoolAPY(hashesIds, { rewardsHistoryLimit: options?.rewardsHistoryLimit })
           );
 
+    // Get pools' stake distribution
+    const stakeDistribution = await this.#cardanoNode.stakeDistribution();
```

Comment:
This should also be cached independently right @IvayloAndonov? Otherwise, it will be called for every pool. This is how it's used in the `NetworkInfoProvider`
https://github.com/input-output-hk/cardano-js-sdk/blob/f5f657789421d0fb186142e6494f0a190d26b4f9/packages/cardano-services/src/NetworkInfo/DbSyncNetworkInfoProvider/DbSyncNetworkInfoProvider.ts#L87-L92

Reply:
Right, should be cached with the default TTL since it's provide live data, good catch.

Reply:
Implemented here https://github.com/input-output-hk/cardano-js-sdk/pull/386/files#diff-5ad8b26faabe50605214aec922ed96428a741e5e07da541676062cec7a693a17R176